table/block_based/block_based_table_reader.cc

```diff
@@ -2111,14 +2111,40 @@ void BlockBasedTable::FullFilterKeysMayMatch(
   if (filter == nullptr || filter->IsBlockBased()) {
     return;
   }
+  uint64_t before_keys = range->KeysLeft();
+  if (before_keys == 0) {
+    // Nothing left to filter
```

Comment:
This shouldn't happen as it means the table reader's MultiGet() was called with 0 keys. Maybe assert here.

Reply:
The caller of FullFilterKeysMayMatch has a prior check on whether the range is empty. Hmm. I also find the skip_filters check in `if (skip_filters || !sst_file_range.empty())` odd.

Reply:
In fact, `sst_file_range.begin()->get_context` conditional on `(skip_filters || !sst_file_range.empty())` would be out of bounds if skip_filters true and sst_file_range empty.